packages/vite/src/node/ssr/runnerImport.ts

```diff
@@ -39,6 +39,9 @@ export async function runnerImport<T>(
           },
         },
       },
+      server: {
+        perEnvironmentStartEndDuringDev: true,
+      },
```

Comment:
this may bring issues in the future, we can add `perEnvironmentStartEndDuringDev: true` to the CSS plugin, no?

Reply:
I wasn't sure about adding `perEnvironmentStartEndDuringDev` to css plugins since currently css plugins tracks states based on top level `ResolvedConfig`, so calling `buildStart` multiple times can end up breaking the state.

Other plugins such as `vite:manifest` and `vite:assets` already use `perEnvironmentStartEndDuringDev`, but those plugins are written based on "per environment state", but that's not the case for css plugins.

I think there is something I misunderstood about `buildStart` in general, and for example, my fix on vite-node https://github.com/vitest-dev/vitest/pull/7480 isn't probably good, so I have to revisit this one too. Probably I'll create an issue first.